### ISSUE TITLE
Dependencies outside of NPM

### ISSUE THREAD

**Issue by Schniz** [rlidwka/sinopia#131]
when I try to use sinopia in a closed network situation, and do my `npm install socket.io`.. it seems that socket.io has a dependency (somehow) of a github repo.

I thought about adding a feature that would override the github dependency in the `package.json` and will download it as if it was a npm dependency.

your thoughts?

**Comment by geekytime** [comment]
> when I try to use sinopia in a closed network situation, and do my npm install socket.io.. it seems that socket.io has a dependency (somehow) of a github repo.

Just in case you didn't realize, you should still be able to configure your local npm settings to go out through your network's proxy server. This will allow you to download npm dependencies through Sinopia, and github dependencies through your proxy server. At least, that worked for me.

Rather than overriding dependencies (which scares me just a little), my team is considering trying to set Sinopia up as our proxy, and having it track (and cache?) github dependencies as they're downloaded, as it does for npm dependencies.

**Comment by Schniz** [comment]
@geekytime, you are correct, but my case is not that common. I can't connect my machine to the npm registry or github. we are, practically, disconnected from the world :cry: 

So overriding dependencies seems legit in this use case. is your case different?
